File: Ass2/t3.cpp
#include <stdio.h>
const char* judgeNum(int number);//Judge if the number is perfect square.
int digitNum(int number);//Check the number of digits.
int main()
{
	int number;
	puts("Please input a positive integer: ");
	scanf("%d", &number);
	if (number <= 0)//Situation when the number is not a positive integer
	{
		puts("Please input a positive integer!");
		return -1;
	}
	printf("The number of digits of %d is %d, %d is %s.\n", number, digitNum(number), number, judgeNum(number));
	return 0;
}
const char* judgeNum(int number)
{
	int i = 1;
	do
	{
		if (i * i == number)
			return "a perfect square";
		i++;
	} while (number >= i * i);
	return "not a perfect square";
}
int digitNum(int number)
{
	int length = 1;//The length of input number
	while (number >= 10)
	{
		number /= 10;
		length++;
	}
	return length;
}

File: Ass2/t3.java
import java.util.Scanner;
public class t3 {
    public static void main(String[] args) {
        int number;
        Scanner scan = new Scanner(System.in);
        number = scan.nextInt();
        if (number <= 0){
            System.out.println("Please input a positive integer.");
            System.exit(-1);
        }
        if (Math.sqrt(number) % 1 == 0)
            System.out.printf("%d is a perfect square, the digit of number of it is %d.\n", number, digitNumber(number));
        else
            System.out.printf("%d is not a perfect square, the digit of number of it is %d.\n", number, digitNumber(number));
    }
    private static int digitNumber(int number){
        int length = 1;
        while (number >= 10){
            number /= 10;
            length++;
        }
        return length;
    }
}